components/ai_chat/core/browser/conversation_driver.cc

```diff
@@ -692,6 +700,24 @@ void ConversationDriver::CleanUp() {
   }
 }
 
+void ConversationDriver::OnModelListUpdated() {
+  for (auto& obs : observers_) {
```

Comment:
We need to re-init the engine if the current model was modified, otherwise active conversations won't get new details, e.g. when first setting up a new custom model users are bound to get something wrong and they won't know existing conversations aren't using the updated setting.

Reply:
added `UpdateModelOption` api to consumers. additionally, wrote a test for this API too.